src/compiler/parser.ts

```diff
@@ -83,7 +83,8 @@ module ts {
                     visitNodes(cbNodes, node.modifiers) ||
                     visitNodes(cbNodes, (<SignatureDeclaration>node).typeParameters) ||
                     visitNodes(cbNodes, (<SignatureDeclaration>node).parameters) ||
-                    visitNode(cbNode, (<SignatureDeclaration>node).type);
+                    visitNode(cbNode, (<SignatureDeclaration>node).type) ||
+                    visitNode(cbNode, (<SignatureDeclaration>node).typePredicate);

```

Comment:
I had a discussion earlier with @JsonFreeman . We came to the conclusion that type predicate info should be stored in a separate property in `Signature`. I thought to keep things consistent  we would store it on a separate property in `SignatureDeclaration`.

In my first implementation of this — I stored it on the type node.